pkg/operations/kubernetesupgrade/upgrader.go

```diff
@@ -557,6 +596,94 @@ func (ku *Upgrader) generateUpgradeTemplate(upgradeContainerService *api.Contain
 	return templateMap, parametersMap, nil
 }
 
+func (ku *Upgrader) getLastVMNameInVMSS(ctx context.Context, resourceGroup string, vmScaleSetName string) (string, error) {
```

Comment:
Is the order guaranteed by the api?
Is this getting the newly created VM by listing all vm and taking the last in the list?

Reply:
Yes, the newly created VM in an VMSS is always the last in the list.